core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BillingProjectOrchestrator.scala

```diff
@@ -210,28 +189,33 @@ class BillingProjectOrchestrator(ctx: RawlsRequestContext,
             )
         }
       billingProfileId <- billingRepository.getBillingProfileId(projectName)
-      projectLifecycle = billingProfileId match {
-        case None    => googleBillingProjectLifecycle
-        case Some(_) => bpmBillingProjectLifecycle
+      (deleteType, projectLifecycle) = billingProfileId match {
+        case None    => (GoogleBillingProjectDelete, googleBillingProjectLifecycle)
+        case Some(_) => (BpmBillingProjectDelete, bpmBillingProjectLifecycle)
       }
       _ <- billingRepository.failUnlessHasNoWorkspaces(projectName)
-      (jobControlId, monitorJobType) <- projectLifecycle.initiateDelete(projectName, ctx)
-      _ <- monitorJobType match {
+      _ <- billingRepository.getCreationStatus(projectName).map { status =>
+        if (!CreationStatuses.terminal.contains(status))
+          throw new BillingProjectDeletionException(
+            ErrorReport(
+              s"Billing project ${projectName.value} cannot be deleted when in status of $status"
+            )
+          )
+      }
+      jobControlId <- projectLifecycle.initiateDelete(projectName, ctx)
+      _ <- deleteType match {
         case GoogleBillingProjectDelete => projectLifecycle.finalizeDelete(projectName, ctx)
-        case _ =>
+        case BpmBillingProjectDelete =>
```

Comment:
Is Scala okay with this being a non-exhaustive match? I would've expected you to need a `_` case that throws here

Reply:
it knows that it's either one or the other, because it's assigned up above, and those are the only two possibilities assigned